Polish the Mentat encyclopedia UI.

pchote opened the pull request:
Before:
<img width="949" alt="Screenshot 2023-10-29 at 15 55 40" src="https://github.com/OpenRA/OpenRA/assets/167819/fa76f1e0-5533-4035-9be0-1fed8bfaf33e">

After:
<img width="735" alt="Screenshot 2023-10-29 at 17 03 48" src="https://github.com/OpenRA/OpenRA/assets/167819/781e893d-b731-4846-966c-004a6b738024">

The new panel width is set to match the mission browser from #21169.

PunkPun commented:
The window was large to make scroll wheel usable https://github.com/OpenRA/OpenRA/pull/19506#issuecomment-1231431497. While this is better than the oldest version I dunno if we want to shrink the window by that much

pchote commented:
I had that comment in mind when coming up with this layout - it is why it is using a (new) standard width with the mission browser, and is more than twice the height of that most compact version.

PunkPun commented on OpenRA.Mods.Common/Traits/Render/RenderSprites.cs:
```diff
@@ -19,6 +19,12 @@
+		public PlayerPaletteOverrideInit(string name)
+			: base(name) { }
+	}
+
```
Shouldn't this be using the same palette overwriting as asset browser?

PunkPun replied:
Also IMO this should be using campaign colors / palettes. It feels wrong to have a color selection here

pchote replied:
Neither the asset browser (which only acts on a single sprite) or the color picker (which relies on a custom actor definition) approaches work here. I added the color picker because the neutral player colour looked bad - but I like the idea using the campaign colours based on selected style/faction.

PunkPun reviewed (commented):
<img width="2560" alt="Screenshot 2023-10-29 at 20 42 22" src="https://github.com/OpenRA/OpenRA/assets/37534529/1ee8fb81-4f21-43b0-9be3-245e2ce36c07">

it definitely feels too small

pchote commented:
Updated: Increased the panel size to match the mission browser, and dropped the color/faction selector. I may revisit the faction selector later, but it requires some more thought to avoid hacks.

PunkPun commented on mods/d2k/chrome/encyclopedia.yaml:
```diff
@@ -49,31 +49,35 @@ Background@ENCYCLOPEDIA_PANEL:
 									Width: PARENT_RIGHT - 40
+									Height: 25
+									VAlign: Top
+									Align: Center
```
Center alignment looks really weird when the title is not in the middle

PunkPun replied:
it's jarring with longer names

pchote replied:
I think it looks good as-is. Do you have a specific suggestion on what to change?

PunkPun replied:
Ah, I know why the miscommunication. With shorter than 15 character names it looks like the title has an indentation from the description, and then indentation gets disregarded by longer names

pchote replied:
I understand what you're saying, but I don't agree. To me, it seems clear that the the title is associated both with the text and the picture, so the exact relationship between the left of the title and the left of the text doesn't matter.

abcdefg30 replied:
Why center the title and not left align it?

dragunoff replied:
I agree it's jarring with a centered title because it seems that it should line-up with the paragraph (sometimes it almost does):
![image](https://github.com/OpenRA/OpenRA/assets/1355810/d63c40e9-dde5-4346-8bc2-242becc87514)
![image](https://github.com/OpenRA/OpenRA/assets/1355810/df5c4d2f-8468-4bd8-b7e0-836c9c58b4ee)

I'd suggest to either left-align it with the paragraph or with the image. The paragraph would probably work better as the image has a lot of air around it.

pchote committed:
Polish the Mentat encyclopedia UI.

dragunoff commented:
This looks OK overall. However I think that the old layout works better as it doesn't have the text alignment issues and the image lives in its own confined space. The scaling is a nice touch though the vehicles look a little too pixelated to my taste. And I'm not requesting changes - just sharing some thoughts.

pchote commented:
It feels like i'm the only one who *isn't* okay with a dialog that is half completely empty space and with disjointed preview vs text 😕

The momentum is against these changes, so closing.

penev92 commented:
I like the proposal. The ability to select a faction is rather important and the color is a nice addition.
And the current UI does have ... alignment/emptiness issues that need to be addressed!
